### ISSUE TITLE
Extraneous space characters inserted with alignment of equal signs turned on, with an if-block

### ISSUE THREAD

**Issue by ribose-jeffreylau** [rubocop-hq/rubocop#6934]
Extra space characters are inserted when force-aligning equal signs, with if-unless-modifier disabled.  Example input:

```ruby
a = if b?
      c.d = :e
    end
```

The above code block showcases an assignment of an if-block, inside which there's another assignment.

### Possibly related issues:

* https://github.com/rubocop-hq/rubocop/issues/5265
* https://github.com/rubocop-hq/rubocop/issues/6069
* https://github.com/rubocop-hq/rubocop/issues/6259

--------

## Expected behavior

Remains unchanged.

```ruby
a = if b?
      c.d = :e
    end
```

## Actual behavior

Scroll to the right and see that it inserted an absurd amount of space characters in between:

```ruby
a                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 = if b?
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      c.d = :e
    end
```

## Steps to reproduce the problem

When given the following Ruby source,

```ruby
a = if b?
      c.d = :e
    end
```

and the following `.rubocop.yml`,

```yaml
Layout/ExtraSpacing:
  ForceEqualSignAlignment: true

Style/IfUnlessModifier:
  Enabled: false
```

run Rubocop with autofix.

## RuboCop version

```console
$ rubocop -V
0.65.0 (using Parser 2.6.0.0, running on ruby 2.5.3 x86_64-darwin18)
```

**Comment by Drenmi** [comment]
This is reproducible on latest `master`. The corruption happens because an infinite correction loop between `Layout/ExtraSpacing` and `Layout/IndentationWidth`. The loop eventually terminates after a configured number of iterations, leaving the code far off to the right.

**Comment by ribose-jeffreylau** [comment]
Stale bot, this problem still exists as of:

```console
$ rubocop -V
0.72.0 (using Parser 2.6.3.0, running on ruby 2.6.3 x86_64-darwin18)
```

**Comment by jfelchner** [comment]
This should be fixed in #7211

**Comment by jfelchner** [comment]
@Drenmi this can be closed.
